app/src/bcsc-theme/features/modal/DeviceInvalidated.tsx

```diff
@@ -20,23 +31,38 @@ export const DeviceInvalidated = (): JSX.Element => {
    * Handles the factory reset operation.
    */
   const handleFactoryReset = useCallback(async () => {
-    const result = await factoryReset({
-      completedNewSetup: true,
-      completedOnboarding: true,
-      nicknames: store.bcsc.nicknames,
-      selectedNickname: store.bcsc.selectedNickname,
-    })
+    if (!caseType) {
+      logger.warn('DeviceInvalidated: caseType is undefined')
+    }
+    const factoryResetParams: Partial<Record<BCSCReason, Record<string, unknown>>> = {
+      // Can add more cases here for different BCSCReason types in the future
+      [BCSCReason.CanceledByAgent]: {
+        completedNewSetup: true,
+        completedOnboarding: true,
+        nicknames: store.bcsc.nicknames,
+        selectedNickname: store.bcsc.selectedNickname,
+      },
+      [BCSCReason.CanceledByUser]: {}, // Empty for a 'new install state'
+    }
+
+    const result = await factoryReset((caseType && factoryResetParams[caseType]) ?? {})
```

Comment:
nit: I think we might be able to remove this casting with the above type change and the previous required type for caseType